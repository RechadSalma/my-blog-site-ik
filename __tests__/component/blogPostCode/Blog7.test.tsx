import { render } from "@testing-library/react";
import Blog7Codes from "../../../components/blogPostCodes/Blog7";

describe("<Blog7/> code for blog7 component", () => {
  const blogSevenComponent = () => render(<Blog7Codes />);

  test("renders the <code> tag", () => {
    const { getByRole } = blogSevenComponent();

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot Blog7 code tag", () => {
    const { container } = blogSevenComponent();

    expect(container).toMatchSnapshot();
  });
});
