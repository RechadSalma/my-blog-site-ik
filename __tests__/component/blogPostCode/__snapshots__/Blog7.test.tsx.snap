// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog7/> code for blog7 component snapshot Blog7 code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div>
        <span
          class="keyword"
        >
          function
        </span>
         capitaliseString(string: string) {
      </div>
      <div
        class="tab1"
      >
        <div>
          <span
            class="keyword"
          >
            return
          </span>
           string
          <span
            class="method"
          >
            .charAt
          </span>
          (
          <span
            class="value"
          >
            0
          </span>
          )
          <span
            class="method"
          >
            .toUpperCase
          </span>
          () + string
          <span
            class="method"
          >
            .slice
          </span>
          (
          <span
            class="value"
          >
            1
          </span>
          );Â 
        </div>
      </div>
      <div>
        };
      </div>
    </code>
  </pre>
</div>
`;
